log-collector/collector.go

```diff
@@ -0,0 +1,198 @@
+package collector
+
+import (
+	"io"
+	"log"
+	"path/filepath"
+	"sync"
+
+	"github.com/coreos/ktestutil/log-collector/pkg/fluentd"
+	"github.com/coreos/ktestutil/log-collector/pkg/local"
+	"github.com/coreos/ktestutil/log-collector/pkg/s3"
+	"github.com/coreos/ktestutil/log-collector/pkg/scp"
+
+	"k8s.io/client-go/kubernetes"
+)
+
+// Output interface describes where the log file is uploaded
+//
+// Put accepts an 'io.ReadSeeker' of the log file and the 'filename' at the destination.
+// Put returns the 'absolute (accessible) location' of the log file at the destination
+// and an error if the file put was unsuccessful
+type Output interface {
+	Put(io.ReadSeeker, string) (string, error)
```

Comment:
i still feel this interface is strange.

if this is about creating a new file and write to it. we should return a handler. coupling the creating + writing together seems not clean.